### PR TITLE
fix: fix bugs with phpstan and psalm

### PR DESCRIPTION
**Pull request by asbiin**
Update phpstan and psalm to fix some errors prior to Laravel 7 migration

### CODE REVIEW

File: app/Http/Resources/Auditlog/AuditLog.php [hunk 1]
```diff
@@ -20,8 +23,8 @@ public function toArray($request)
             'id' => $this->id,
             'object' => 'auditlog',
             'author' => ($this->author) ? [
-                'id' => $this->id,
-                'name' => $this->name,
+                'id' => $this->author->id,
```
Comment (asbiin) [inline app/Http/Resources/Auditlog/AuditLog.php]: This was a real bug I guess

File: app/Http/Resources/Debt/Debt.php [hunk 2]
```diff
@@ -23,7 +26,7 @@ public function toArray($request)
             'in_debt' => $this->in_debt,
             'status' => $this->status,
             'amount' => $this->amount,
-            'amount_with_currency' => MoneyHelper::format($this->amount),
+            'amount_with_currency' => MoneyHelper::format((int) $this->amount),
```
Comment (asbiin) [inline app/Http/Resources/Debt/Debt.php]: amount should be stored as int

File: app/Models/Account/Activity.php [hunk 3]
```diff
@@ -131,7 +131,7 @@ public function getSummary()
      */
     public function getTitle()
     {
-        return $this->type ? $this->type->key : null;
+        return $this->type ? $this->type->translation_key : null;
```
Comment (asbiin) [inline app/Models/Account/Activity.php]: That was a real bug I guess
@djaiss

File: app/Services/Account/Settings/ResetAccount.php [hunk 4]
```diff
@@ -58,7 +58,7 @@ private function destroyCompanies(Account $account)
     {
         $companies = $account->companies;
         foreach ($companies as $company) {
-            $company->delete;
+            $company->delete();
```
Comment (asbiin) [inline app/Services/Account/Settings/ResetAccount.php]: That's a real bug

File: app/Services/Account/Settings/ResetAccount.php [hunk 5]
```diff
@@ -72,7 +72,7 @@ private function destroyDays(Account $account)
     {
         $days = $account->days;
         foreach ($days as $day) {
-            $day->delete;
+            $day->delete();
```
Comment (asbiin) [inline app/Services/Account/Settings/ResetAccount.php]: That's a real bug

File: app/Services/Account/Settings/ResetAccount.php [hunk 6]
```diff
@@ -86,7 +86,7 @@ private function destroyPlaces(Account $account)
     {
         $places = $account->places;
         foreach ($places as $place) {
-            $place->delete;
+            $place->delete();
```
Comment (asbiin) [inline app/Services/Account/Settings/ResetAccount.php]: That's a real bug

File: app/Services/Account/Settings/ResetAccount.php [hunk 7]
```diff
@@ -127,7 +127,7 @@ private function destroyJournalEntries(Account $account)
     {
         $entries = $account->entries;
         foreach ($entries as $entry) {
-            $entry->delete;
+            $entry->delete();
```
Comment (asbiin) [inline app/Services/Account/Settings/ResetAccount.php]: That's a real bug

**Comment by djaiss** [comment]
I’m regis and i approve this PR!

File: app/Http/Resources/Auditlog/AuditLog.php (hunk 1 above)
Comment (djaiss) [reply app/Http/Resources/Auditlog/AuditLog.php]: Yes it was !!!!!

File: app/Http/Resources/Debt/Debt.php (hunk 2 above)
Comment (asbiin) [reply app/Http/Resources/Debt/Debt.php]: See #3761

**Pull request merged**